data/blog/webview2.md

```diff
@@ -0,0 +1,108 @@
+Electron has [comprehensive security documentation](https://www.electronjs.org/docs/tutorial/security) on choosing your security model.
+WebView2 also has [security best practices](https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/security).
+
+The Electron source is maintained and available on GitHub.
+Applications can modify can build their own _brands_ of Electron.
+The WebView2 source is not available on GitHub.
+
+Quick Summary:
+
+|                                           | Electron        | WebView2                     |
+| ----------------------------------------- | --------------: | ---------------------------: |
+| Build Dependency                          | Chromium        | Edge                         |
+| Source Available on GitHub                | Yes             | No                           |
+| Shares Edge/Chrome DLLs                   | No              | No                           |
+| Shared Runtime Between Applications       | No              | Optional                     |
+| Application APIs                          | Yes             | No                           |
+| Node.js                                   | Yes             | No                           |
+| Sandbox                                   | Optional        | Always                       |
+| Requires a Separate Application Framework | No              | Yes                          |
+| Supported Platforms                       | Mac, Win, Linux | Win (Mac on roadmap)         |
+
+## Performance Discussion
+
+Now the big question:
+_Will your app use 50% less memory by switching between one embedding of Chromium and another?_
+Unfortunately not, but that would have been rad.
+
+We created the [scaffolding to compare some simple apps built using Electron, WPF+WebView2, and .NET+WebView2](https://github.com/crossplatform-dev/xplat-challenges/blob/main/results.md).
+Our goal was to find any egregious differences, rather than deep-dive on micro-benchmarks.
+The source code and detailed results are available in the [xplat-challenges](https://github.com/crossplatform-dev/xplat-challenges) repository,
+so you’re free to play around with the frameworks yourself.
+
+### A Note About Inter-Process Communication (IPC)
+
+In Chromium, the browser process acts as an IPC broker between sandboxed renderers and the rest of the system.
+While Electron allows unsandboxed render processes, many apps choose to enable the sandbox for added security.
+WebView2 always has the sandbox enabled, so for most Electron and WebView2 apps IPC can heavily impact overall performance.
+
+Even though Electron and WebView2 have a similar process models, their underlying IPC implementation is different.
+Communicating between JavaScript and C++ or C# requires [marshalling](https://en.wikipedia.org/wiki/Marshalling_(computer_science)),
+most commonly into a JSON string. This is an expensive operation and IPC bottlenecks can negatively impact performance.
```

Comment:
The bottleneck of most of the IPC call is not from data transfer, many of the IPC call sends little amount of data, there are many other potentially more important factors such as async nature of the call, call back, mojo interaction. It feels we are over emphasize data transfer impact for the speed of the IPC call here.

Reply:
I don't think this implies the overhead is data transfer, but data-serialization and parsing. That has been our experience, and it echoes my experience from node.js. IPC has been a repeated source of confusion in our experience with many applications, and many of them send large amounts of data via IPC.

Synchronous IPC has been an issue, which plagues the remote module, but I can't think of an example off the top of my head where async is causing significant problems.